From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: celenity <celenity@celenity.dev>
Date: Sun, 05 Jul 2026 20:19:20 +0000
Subject: [PATCH] Gecko - Remove privileged add-on signing status

Prevents Mozilla add-ons from returning the 'PRIVILEGED' and 'SYSTEM' add-on signing states.

Signed-off-by: celenity <celenity@celenity.dev>
---
diff --git a/toolkit/mozapps/extensions/internal/XPIInstall.sys.mjs b/toolkit/mozapps/extensions/internal/XPIInstall.sys.mjs
index 358fcce396431..ddd8dc656b774 100644
--- a/toolkit/mozapps/extensions/internal/XPIInstall.sys.mjs
+++ b/toolkit/mozapps/extensions/internal/XPIInstall.sys.mjs
@@ -873,13 +873,13 @@ function getSignedStatus(aRv, aCert, aAddonID) {
         return AddonManager.SIGNEDSTATE_BROKEN;
       }
 
-      if (aCert.organizationalUnit == "Mozilla Components") {
+/*      if (aCert.organizationalUnit == "Mozilla Components") {
         return AddonManager.SIGNEDSTATE_SYSTEM;
       }
 
       if (aCert.organizationalUnit == "Mozilla Extensions") {
         return AddonManager.SIGNEDSTATE_PRIVILEGED;
-      }
+      } */
 
       return /preliminary/i.test(aCert.organizationalUnit)
         ? AddonManager.SIGNEDSTATE_PRELIMINARY
-- 
